### PR TITLE
[Merged by Bors] - Add Nethermind integration tests

### PR DESCRIPTION
**Pull request by macladson**
## Proposed Changes

Extend the current Geth merge integration tests to support Nethermind.

### CODE REVIEW

File: testing/execution_engine_integration/src/nethermind.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    // Update the branch
+    assert!(build_utils::update_branch(&repo_dir, NETHERMIND_BRANCH));
+
+    // Build nethermind
+    build_utils::check_command_output(build_result(&repo_dir), "dotnet build failed");
+
+    // Build nethermind a second time to enable Merge-related features.
+    // Not sure why this is necessary.
+    build_utils::check_command_output(build_result(&repo_dir), "dotnet build failed");
```
Comment (macladson) [inline testing/execution_engine_integration/src/nethermind.rs:46]: Without this 2nd build, Nethermind errors with:
```
{"jsonrpc":"2.0","error":{"code":-32601,"message":"Method engine_exchangeTransitionConfigurationV1 is not supported"},"id":1}
```
so it seems like the `engine` module isn't being built the first time.

There may be some trickiness with Git not properly checking out the branch, but I'm not sure why it works for Geth.
Happy to hear suggestions.

Comment (pawanjay176) [reply testing/execution_engine_integration/src/nethermind.rs:46]: Checked with the Nethermind folks. The issue seems to be that the first build doesn't copy the `plugins` folder in bin. 
The engine api doesn't get enabled without the `merge` plugin.

Building twice seems to build the `plugins` folder which enables the engine api. So I think this is probably the cleanest way to get the tests working until it's fixed on Nethermind.

File: testing/execution_engine_integration/src/build_utils.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        .status
+        .success()
+}
+
+pub fn check_command_output(output: Output, failure_msg: &'static str) {
+    if !output.status.success() {
+        let stdout = String::from_utf8_lossy(&output.stdout);
+        let stderr = String::from_utf8_lossy(&output.stderr);
+
+        dbg!(stdout);
```
Comment (pawanjay176) [inline testing/execution_engine_integration/src/build_utils.rs]: Should remove these before bors

File: testing/execution_engine_integration/src/geth.rs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    // Checkout the correct branch
+    assert!(build_utils::checkout_branch(&repo_dir, GETH_BRANCH));
+
+    // Update the branch
+    assert!(build_utils::update_branch(&repo_dir, GETH_BRANCH));
+
+    // Build geth
+    let make_result = Command::new("make")
```
Comment (pawanjay176) [inline testing/execution_engine_integration/src/geth.rs]: Can maybe extract this into a separate function like nethermind build?

**Review by pawanjay176 — APPROVED** [review]
Works perfectly for me. :ok_hand: 
I think the workaround for the nethermind issue is good enough for now.

File: testing/execution_engine_integration/src/build_utils.rs (hunk 2 above)
Comment (macladson) [reply testing/execution_engine_integration/src/build_utils.rs]: I think these are useful for debugging. I imagine these tests will break quite frequently as the spec changes, the execution clients update, etc, so they will probably be useful in the future. I'll put them behind the `SUPPRESS_LOGS` bool for now.

Comment (pawanjay176) [reply testing/execution_engine_integration/src/build_utils.rs]: Sorry I thought it was used for temporary debugging :sweat_smile:

**Review by paulhauner — APPROVED** [review]
Nice one! :tada:

**Pull request closed without merging**